code/__HELPERS/lists.dm

```diff
@@ -71,6 +71,32 @@ proc/isemptylist(list/list)
 			return 1
 	return 0
 
+/proc/is_type_in_oview(var/type, var/dist = 0, var/center = src)
+	if (!ispath(type))
+		CRASH("Not a valid type in 'is_type_in_oview()'")
+	if (!isnum(dist))
+		CRASH("Not a valid dist in 'is_type_in_oview()'")
+	if (!isloc(center))
+		CRASH("Not a valid center in 'is_type_in_oview()'")
+	var/list/atoms = oview(dist, center)
+	for (var/atom/A in atoms)

```

Comment:
This is inefficient.

`var/atom/A` executes an `istype(A, /atom)` check for every object in the list. But we're executing another, way more restrictive check just below as well.

Remove the /atom child here, and simply run `for (var/A in atoms)`. It'll be faster, with no issues.